file.go

```diff
@@ -216,7 +221,8 @@ func readMetadata(pefile *pe.File, rva uint32) (string, []*Heap, error) {
 	}
 
 	// parse stream headers.
-	heaps := make([]*Heap, streamsCount)
+	var streams []*Heap
+	streamNames := make(map[string]struct{})
```

Comment:
Why is this no longer using `streamsCount` when allocating? Don't `streams` and `streamNames` have a consistent max size in the successful case, which is the one that makes sense to optimize?

Reply:
I noticed the commit message "limit allocations based on user inputs". I would think that a malicious file would pick different names for each stream... but I suppose this would reduce allocations in some cases.

Reply:
Well, common case is to have just the 5 streams defined in ecma-335, so I'll reserve some capacity for it.